llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1528,6 +1525,13 @@ bool LoopVectorizationLegality::canVectorize(bool UseVPlanNativePath) {
     else
       return false;
   }
+  if (Result) {
+    LLVM_DEBUG(dbgs() << "LV: We can vectorize this loop"
```

Comment:
Can this be kept at the original place? the check above is a profitability check, we can still vectorize the loop

Might be good to have a test case

Reply:
Done. But I think the meaning attached to "false" returned from this function is slightly misleading. If "false" is returned through the last instance then the loop is vectorizable but not profitable. For other instances, it is not vectorizable at all. I think the name of the function should be `canVectorizeAndProfitable`.